### ISSUE TITLE
Support for GeoJSON-Data

### ISSUE THREAD

**Issue by Edefritz** [SINTEF-9012/PruneCluster#15]
Hi everyone,
seems like a really nice project. I was just wondering if it is possible to use GeoJSON-Data as Input.

Thanks!

**Comment by adambarthelson** [comment]
So has anyone touched this in the passed 5 months?

![](http://i.imgur.com/s1fPqX1.gif)

I was thinking about giving it a go.

**Comment by rwwagner90** [comment]
@adambarthelson are you using geoJson for markers or shapes? If it's just markers, you don't need to use geoJson. If it's shapes, I can help you out with the way I supported shapes.

**Comment by adambarthelson** [comment]
You're right, at the time I figured using GeoJSON would be more real-time friendly, but have since found a better approach to the problem. P.S. Fairfax, VA FTW.

**Comment by rwwagner90** [comment]
@adambarthelson so markers will work for you then? Haha represent!

**Comment by tohodakilla** [comment]
would love to see this feature

**Comment by maxvonhippel** [comment]
I think a simple example with points (no polygons or other fancy stuff) being clustered from a .json or .geojson file would be very helpful.  It's not that hard to do, but it would make the code that much easier for a lot of people to jump into.  Also, this repo to some extent "competes" (insofar as open-source coding can be called "competitive", so not sure if that's really the right word) against Mapbox's SuperCluster, and SuperCluster does have a (not necessarily totally functional) demo with JSON.

**Comment by maxvonhippel** [comment]
I just managed to make it work, with a JSON file containing over 3,000,000 features.  It takes about a minute to render on my MacBook pro but is completely smooth once rendered.

**Comment by maxvonhippel** [comment]
OK I just created a pull request with this feature.  Please check it out!  Thanks :) https://github.com/SINTEF-9012/PruneCluster/pull/135

**Comment by yellowiscool** [comment]
Thanks @maxvonhippel for the example and the dataset. Nevertheless,

PruneCluster should still support GeoJSON with a simple API though. Hopefully someday :-)

**Comment by maxvonhippel** [comment]
Thanks for merging!  This was my first ever contribution to an open-source project :D but yes I agree, an API-level GeoJSON functionality would eventually be ideal.

**Comment by iBrian71** [comment]
I would like to see GEOJson support as well.  I love the PruneCluster features.
Being able to add layers to prunecluster would also be beneficial, similar to markercluster.

**Comment by bre7** [comment]
@rwwagner90 Any chance you can share a snipet for your GeoJSON + shapes example (poly) ?

**Comment by koren88i** [comment]
is still this a thing? I would really want to use prunecluster to cluster polygons. 
use .getcenter for the clustering but show polygon and not icon (marker) when showing the shape.

is this possible?
